src/sql/src/plan/statement/ddl.rs

```diff
@@ -1265,19 +1273,42 @@ fn kafka_sink_builder(
             if seed.is_some() {
                 bail!("SEED option does not make sense with sinks");
             }
-            (url.parse::<Url>()?, normalize::options(&with_options))
+            let schema_registry_url = url.parse::<Url>()?;
+            let ccsr_with_options = normalize::options(&with_options);
+            let ccsr_config = kafka_util::generate_ccsr_client_config(
+                schema_registry_url.clone(),
+                &config_options,
+                ccsr_with_options,
+            )?;
+
+            let schema_generator = AvroSchemaGenerator::new(
+                key_desc_and_indices
+                    .as_ref()
+                    .map(|(desc, _indices)| desc.clone()),
+                value_desc.clone(),
+                consistency_topic.is_some(),
+            );
+            let value_schema = schema_generator.value_writer_schema().to_string();
+            let key_schema = schema_generator
+                .key_writer_schema()
+                .map(|key_schema| key_schema.to_string());
+
+            KafkaSinkFormat::Avro {
+                schema_registry_url,
+                key_schema,
+                value_schema,
+                ccsr_config,
+            }
+        }
+        _ => {
+            // TODO@jldlaughlin: if format is not Avro with schema, fail if
```

Comment:
This might sound weird, but I think the combination of JSON for data output and Avro for the consistency topic can make sense.

Right now, the consistency topic that we write out is not really useful for external consumers. It's really only used for the exactly-once sink to record the latest written timestamp/transaction.

Reply:
If we want JSON data output + JSON consistency topic, we also need to update the bootstrapping logic that gets the latest timestamp from the topic: https://github.com/MaterializeInc/materialize/blob/5b92294f57235a00e92634ea8961c4fad4db27d2/src/coord/src/sink_connector.rs#L68

Which, doable, but a hassle. 🙈

Reply:
Ah, got it! Yeah, I think JSON data output + JSON consistency topic can definitely be a V2+.